### PR TITLE
Update DuckDB to latest

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Comment by majetideepak** [comment]
@mbasmanova I amended my commit author as `svcscm svcscm <svcscm@fb.com>` with the idea to attribute all vendored/generated code to a single not-real user. However, I am not sure if that will get reflected when we merge this :).
See commit details for author details.

**Comment by majetideepak** [comment]
I also verified that this PR fixes https://github.com/facebookincubator/velox/issues/895.
I will add that test in a separate PR.

**Comment by majetideepak** [comment]
I will look into the build failures.

**Comment by majetideepak** [comment]
A couple of changes due to https://github.com/duckdb/duckdb/pull/2869

**Review by mbasmanova — APPROVED** [review]
@majetideepak Thank you, Deepak.

**Comment by mbasmanova** [comment]
> @mbasmanova I amended my commit author as `svcscm svcscm <svcscm@fb.com>` with the idea to attribute all vendored/generated code to a single not-real user. However, I am not sure if that will get reflected when we merge this :). See commits for author details.

Let's see what happens when we land this. I don't think commit metadata is used when importing the PR into fbcode though.

**Comment by majetideepak** [comment]
I will look into the test failures

File: velox/duckdb/conversion/DuckParser.cpp [hunk 1]
```diff
@@ -384,7 +385,9 @@ std::shared_ptr<const core::IExpr> parseExpr(ParsedExpression& expr) {
 } // namespace
 
 std::shared_ptr<const core::IExpr> parseExpr(const std::string& exprString) {
-  auto parsedExpressions = Parser::ParseExpressionList(exprString);
+  ParserOptions options;
+  options.preserve_identifier_case = false;
```
Comment (majetideepak) [inline velox/duckdb/conversion/DuckParser.cpp:389]: Required due to changes from https://github.com/duckdb/duckdb/pull/2919

**Comment by aditi-pandit** [comment]
> _No description provided._

Hi Deepak,

  What is the process for doing this merge ? Is that documented somewhere for others reference ?

**Comment by majetideepak** [comment]
@aditi-pandit, here are the steps https://github.com/facebookincubator/velox/tree/main/velox/duckdb

**Comment by aditi-pandit** [comment]
> @aditi-pandit, here are the steps https://github.com/facebookincubator/velox/tree/main/velox/duckdb

Thanks

**Pull request closed without merging**